libnavui-androidauto/src/main/java/com/mapbox/androidauto/screencoordinator/MapboxScreenManager.kt

```diff
@@ -0,0 +1,237 @@
+package com.mapbox.androidauto.screencoordinator
+
+import androidx.annotation.UiThread
+import androidx.car.app.CarContext
+import androidx.car.app.Screen
+import androidx.car.app.ScreenManager
+import androidx.car.app.Session
+import androidx.car.app.model.Template
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.androidauto.internal.logAndroidAuto
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.launch
+import kotlin.jvm.Throws
+
+/**
+ * The Mapbox Navigation Android Auto SDK is prepared with a default experience. This object allows
+ * you to customize the experience to meet your needs.
+ */
+class MapboxScreenManager {
+    private var carContext: CarContext? = null
+    private var screenManager: ScreenManager? = null
+    private var lifecycleOwner: LifecycleOwner? = null
+    private val screenFactoryMap = mutableMapOf<String, MapboxScreenFactory>()
+    private lateinit var coroutineScope: CoroutineScope
+
+    private val lifecycleObserver: DefaultLifecycleObserver = object : DefaultLifecycleObserver {
+        override fun onCreate(owner: LifecycleOwner) {
+            val carContext = requireCarContext()
+            screenManager = carContext.getCarService(ScreenManager::class.java)
+        }
+
+        override fun onDestroy(owner: LifecycleOwner) {
+            onDestroy()
+        }
+
+        override fun onStart(owner: LifecycleOwner) {
+            coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
+            coroutineScope.launch {
+                screenEvent
+                    .filterNotNull()
+                    .collect {
+                        onCarScreenEvent(it)
+                        logAndroidAuto("MapboxScreenManager screen changed $it")
+                    }
+            }
+        }
```

Comment:
Noticed that when you press home, then reopen the app, it will push or replace the last screen again. So you may have two or even more feedback screens opened, for example.

Reply:
https://github.com/mapbox/mapbox-navigation-android/blob/8300e2f8b4213d0b6d7d047dc26b6c9420190832/libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/MainScreenManager.kt#L66-L68

Oh right, old way checks the top

Reply:
This calls for tests because it's going to have a few issues and cases come up.

Reply:
Actually after testing it, there is no need to attach this to the STARTED or RESUMED state.

Hooking it up to the CREATED works just fine

Reply:
ok it's updated